extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-backend-module.ts

```diff
@@ -43,7 +47,8 @@ export default new ContainerModule((bind, unbind, isBound, rebind) => {
   bind(HostedPluginMapping).toSelf().inSingletonScope();
   bind(MetadataProcessor).to(RemoteMetadataProcessor).inSingletonScope();
   bind(PluginReaderExtension).toSelf().inSingletonScope();
-
+  rebind(HostedPluginReader).to(PluginReaderExtension).inSingletonScope();
```

Comment:
shouldn't it be toService(PluginReaderExtension) to use the same instance ? (and then we don't need to cast pluginReaderExtension as any type to call setRemotePluginConnection on it ?
else it looks odd

Reply:
I tried that and it did didn't work. T.b.h: I have no idea what "toService()" really does and inversify documentation didn't enlighten me.

Reply:
it's to bind the same instance under different interfaces (and not having two different singletons intances)
Here it looks like code smell, we have two different instances of the same class bound on different interfaces.